app/Controller/ServersController.php

```diff
@@ -1266,6 +1266,88 @@ private function __checkVersion()
         }
     }
 
+    public function idTranslator() {
+
+        // The id translation feature is limited to people from the host org
+        if (!$this->_isSiteAdmin() && $this->Auth->user('org_id') != Configure::read('MISP.host_org_id')) {
+            throw new MethodNotAllowedException(__('You don\'t have the required privileges to do that.'));
+        }
+
+        //We retrieve the list of remote servers that we can query
+        $options = array();
+        $options['conditions'] = array("pull" => true);
+        $servers = $this->Server->find('all', $options);
+
+        // We generate the list of servers for the dropdown
+        $displayServers = array();
+        foreach($servers as $s) {
+            $displayServers[] = array('name' => $s['Server']['name'],
+                                      'value' => $s['Server']['id']);
+        }
+        $this->set('servers', $displayServers);
+
+        if ($this->request->is('post')) {
+            $remote_events = array();
+            if(!empty($this->request->data['Event']['uuid']) &&  $this->request->data['Event']['local'] == "local") {
+                $local_event = $this->Event->fetchSimpleEvent($this->Auth->user(), $this->request->data['Event']['uuid']);
+            } else if (!empty($this->request->data['Event']['uuid']) && $this->request->data['Event']['local'] == "remote" && !empty($this->request->data['Server']['id'])) {
+                //We check on the remote server for any event with this id and try to find a match locally
+                $conditions = array('AND' => array('Server.id' => $this->request->data['Server']['id'], 'Server.pull' => true));
+                $remote_server = $this->Server->find('first', array('conditions' => $conditions));
+                if(!empty($remote_server)) {
+                    try {
+                        $remote_event = $this->Event->downloadEventFromServer($this->request->data['Event']['uuid'], $remote_server, null, true);
+                    } catch (Exception $e) {
+                        $this->Flash->error(__("Issue while contacting the remote server to retrieve event information"));
+                        $this->log($e->getMessage(), LOG_NOTICE);
```

Comment:
You can use `$this->logException`, so the whole exception with backtrace will be logged.